File: 9-palindrome-number/palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x < 0) {
            return false;
        }

        boolean ans = false;
        int check = 0;
        int num = x;
        while(x > 0) {
            int digit = x % 10;
            check = check * 10 + digit;
            x = x / 10;
        }

        if(check == num) {
            return true;
        }

        return false;
    }
}

File: 9-palindrome-number/palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """

        if x < 0:
            return False

        temp = x
        num = 0
    
        while x > 0:
            last = x % 10
            num = last + num * 10
            print("checking num", num)
            x = x // 10

        return temp == num